### PR TITLE
Fix strategies.

### PR DESCRIPTION
**Pull request by KaiPrince**
Fix Crypter get_strategy and convert_ext. 
Maintenance & documentation fixes.

### CODE REVIEW

**Review by SamueLagunju — APPROVED** [review]
Greatest Pull Request of all time.

File: crypto/crypter.py [hunk 1]
```diff
@@ -4,34 +4,33 @@
...
-        strategies = [SeanStrategy(), RubikStrategy(), DocStrategy()]
-
-        return strategies
+    def __init__(
+        self,
+        strategy_factory: Callable[[], List[Strategy]] = make_strategies,
+        fileio=_fileio,
+    ):
+        self.strategies = strategy_factory()
+        self.fileio = fileio
```
Comment (KaiPrince) [inline crypto/crypter.py:28]: All hail dependency injection.

File: tests/test_crypter.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    # Arrange
+    contents = "this is a string"
+    # ..mock Strategy
+    mock_strategy = mocker.MagicMock()
+    mock_strategy.get_supported_types.return_value = [ExtPair(".txt", ".crp")]
+    # ..mock fileio
+    mock_file_reader = mocker.MagicMock()
+    mock_file_reader.read_file.return_value = contents
+    # ..make crypter
+    crypter = Crypter(strategy_factory=lambda: [mock_strategy], fileio=mock_file_reader)
```
Comment (KaiPrince) [inline tests/test_crypter.py:37]: Because it makes this so much easier.

Comment (SamueLagunju) [reply tests/test_crypter.py:37]: Makes a lot of sense. Maybe I won't be so scared of pytest now.

**Pull request merged**